osu.Game/Overlays/News/Sidebar/MonthPanel.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Graphics.Containers;
+using osuTK;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics;
+using System.Linq;
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Sprites;
+
+namespace osu.Game.Overlays.News.Sidebar
+{
+    public class MonthPanel : CompositeDrawable
+    {
+        public readonly BindableBool IsOpen = new BindableBool();
+
+        private readonly FillFlowContainer postsFlow;
+
+        public MonthPanel(List<APINewsPost> posts)
+        {
+            Width = 160;
+            AutoSizeDuration = 250;
+            AutoSizeEasing = Easing.OutQuint;
+            InternalChild = new FillFlowContainer
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Direction = FillDirection.Vertical,
+                Spacing = new Vector2(0, 5),
+                Children = new Drawable[]
+                {
+                    new DropdownButton(posts[0].PublishedAt)
+                    {
+                        IsOpen = { BindTarget = IsOpen }
+                    },
+                    postsFlow = new FillFlowContainer
+                    {
+                        RelativeSizeAxes = Axes.X,
+                        AutoSizeAxes = Axes.Y,
+                        Direction = FillDirection.Vertical,
+                        Spacing = new Vector2(0, 5),
+                        Children = posts.Select(p => new PostButton(p)).ToArray()
+                    }
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            IsOpen.BindValueChanged(open =>
+            {
+                ClearTransforms();
+
+                if (open.NewValue)
+                {
+                    AutoSizeAxes = Axes.Y;
+                    postsFlow.FadeIn(250, Easing.OutQuint);
+                }
+                else
+                {
+                    AutoSizeAxes = Axes.None;
+                    this.ResizeHeightTo(15, 250, Easing.OutQuint);
+
+                    postsFlow.FadeOut(250, Easing.OutQuint);
+                }
+            }, true);
+
+            // First state change should be instant.
+            FinishTransforms();
+            postsFlow.FinishTransforms();
```

Comment:
```suggestion
            FinishTransforms(true);
```

Reply:
Also this won't work on the autosize transforms of `MonthPanel`, they're added way later on at the end of `CompositeDrawable.UpdateSubTree`, there may be a solution for this case somewhere around the game, but the only way I can think is overriding `UpdateAfterAutoSize` and call once instead, or something around that.